internal/app/networkd/pkg/address/dhcp.go

```diff
@@ -129,7 +130,14 @@ func (d *DHCP) Resolvers() []net.IP {
 func (d *DHCP) Hostname() string {
 	// Truncate the returned hostname to only return
 	// the actual host entry
-	return strings.Split(d.Ack.HostName(), ".")[0]
+	hostname := strings.Split(d.Ack.HostName(), ".")[0]
```

Comment:
since we do this split later, should we return d.ack.hostname in full if d.ack.domainname == ""?

Reply:
Good question.  That sounds reasonable.